src/s3-update.ts

```diff
@@ -26,20 +28,23 @@ export async function s3Update(options: S3UpdateOptions): Promise<void> {
   // However, branches are not intended for production use, so the occasional broken
   // branch does not seem worth fixing.
 
-  const { deployPath, version, branch, bucket, prefix, topBranchesJSON } = options;
+  const { deployPath, version, branch, bucket, prefix, topBranchesJSON, localFolder} = options;
 
   const topLevelS3Url = `s3://${bucket}/${prefix}`;
   const deployS3Url = `${topLevelS3Url}/${deployPath}`;
   const maxAgeSecs = version ?  MAX_AGE_VERSION_SECS : MAX_AGE_BRANCH_SECS;
 
-
   const excludes = `--exclude "index.html" --exclude "index-top.html"`;
   const cacheControl = `--cache-control "max-age=${maxAgeSecs}"`;
-  await exec.exec(`aws s3 sync ./dist ${deployS3Url} --delete ${excludes} ${cacheControl}`);
+  await exec.exec(`aws s3 sync ./${localFolder} ${deployS3Url} --delete ${excludes} ${cacheControl}`);
 
   const noCache = `--cache-control "no-cache, max-age=0"`;
-  await exec.exec(`aws s3 cp ./dist/index.html ${deployS3Url}/ ${noCache}`);
-  await exec.exec(`aws s3 cp ./dist/index-top.html ${deployS3Url}/ ${noCache}`);
+  await exec.exec(`aws s3 cp ./${localFolder}/index.html ${deployS3Url}/ ${noCache}`);
+
+  const indexTopPath = `${localFolder}/index-top.html`;
+  if (fs.existsSync(indexTopPath)) {
+    await exec.exec(`aws s3 cp ./${indexTopPath} ${deployS3Url}/ ${noCache}`);
+  }
```

Comment:
The use of the `indexTopPath` local variable in line 44 makes the two execs in lines 42 and 46 seem more different than they are. Adding an analogous `indexPath` before line 42 would make the similarities more apparent.

Reply:
Sounds good.